bitnami/solr/templates/_helpers.tpl

```diff
@@ -106,13 +106,20 @@ Return Solr admin password
 {{- end -}}
 
 {{/*
-Return the proper Zookeeper host
+Return proper Zookeeper hosts
 */}}
-{{- define "solr.zookeeper.host" -}}
+{{- define "solr.zookeeper.hosts" -}}
 {{- if .Values.externalZookeeper.servers -}}
     {{- include "common.tplvalues.render" (dict "value" (join "," .Values.externalZookeeper.servers) "context" $) -}}
 {{- else -}}
-    {{- printf "%s:%d" (include "solr.zookeeper.fullname" .) (int .Values.zookeeper.containerPorts.client) -}}
+    {{- $zookeeperList := list -}}
+    {{- $releaseNamespace :=  default .Release.Namespace .Values.zookeeper.namespaceOverride -}}
+    {{- $clusterDomain := .Values.clusterDomain -}}
+    {{- $zookeeperFullname := include "solr.zookeeper.fullname" . -}}
+    {{- range $e, $i := until (int .Values.zookeeper.replicaCount) -}}
+        {{- $zookeeperList = append $zookeeperList (printf "%s-%d.%s-headless.%s.svc.%s:%d" $zookeeperFullname $i $zookeeperFullname $releaseNamespace $clusterDomain (int $.Values.zookeeper.containerPorts.client))  -}}
```

Comment:
Can you explain an example of what this connection string will look like? Looks like you are addressing each zookeeper replica pod "inside" the headless service, I'm not familiar with this style.

Reply:
Yes, that's the aim of that piece of code. Using the default values, the result will be a list like this one:
```
[ 'solr-zookeeper-0.solr-zookeeper-headless.default.svc.cluster.local:2181', 'solr-zookeeper-1.solr-zookeeper-headless.default.svc.cluster.local:2181', 'solr-zookeeper-2.solr-zookeeper-headless.default.svc.cluster.local:2181' ]
```

NOTE: Pods governed by a StatefulSet + Headless service keep a [stable network ID](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#stable-network-id)